### ISSUE TITLE
User Feedback

### ISSUE THREAD

**Issue by MakingMadness** [alondmnt/joplin-plugin-jarvis#4]
Can we have some feedback on what's happening after a request is sent? An indicator that we're awaiting a response, and an error message if there is one?

For instance, I might press Ctrl+Shift+C and the border of the textarea starts glowing for the duration of whatever timeout period we're waiting. It gets an error message so the textarea stops glowing and the message is displayed to the user.

**Comment by alondmnt** [comment]
I agree that some feedback could be nice. I'll see what I can do in terms of visuals.

Currently, error messages are displayed in a dialog whenever something happens (error on server, bad API key, prompt errors, etc.). I believe this includes timeouts, however the maximal waiting period is very long by default.

I can implement a timeout mechanism (did it for the VS Code version of Jarvis at some point), where you can set the timeout period in the settings, and get an error dialog whenever time runs out. Does this sound helpful?

**Comment by MakingMadness** [comment]
Sure, that sounds great. It would be nice to not have to consider these things causing the delay instead:

- "Did I press those keys correctly?"
- Problems with the plugin
- Problems with Joplin
- Problems with your pc, like freezes.

And instead just see that something is clearly happening and everything is good.
